charts/consul/templates/connect-inject-deployment.yaml

```diff
@@ -83,6 +83,13 @@ spec:
             - |
               CONSUL_FULLNAME="{{template "consul.fullname" . }}"
 
+              {{- if .Values.dns.enabled }}
+              consul_fullname_with_underscore=$(echo $CONSUL_FULLNAME | tr '-' '_')
+              dns_var_name=${consul_fullname_with_underscore}_dns_service_host
+              dns_var_name_in_caps=$(echo $dns_var_name | tr 'a-z' 'A-Z')
+              export dns_ip=$(eval echo \$${dns_var_name_in_caps})
```

Comment:
here, we are essentially templating the name of the environment variable that will have the Consul DNS IP. If `CONSUL_FULLNAME` is `consul-consul`, then the env var would be `CONSUL_CONSUL_DNS_SERVICE_HOST`, so we need to turn dashes into underscores, then capitalize all the letters. Then to dereference the variable once we've come up with the name, we use the `eval echo \$$` bit to do that

Reply:
in discussion, we think we could move this to inject-connect